src/contexts/favBlog/FavBlogContext.js

```
@@ -0,0 +1,58 @@
+import React, { createContext, useContext, useReducer } from "react";
+
+// create context
+const FavBlogPostsContext = createContext();
+
+//The reducer function contains your custom state logic
+//and the initialState can be a simple value but generally will contain an object.
+const initialState = { favBlogs: [] };
+
+function favBlogReducer(state, action) {
+  const { type, payload } = action;
+  const { favBlogs } = state;
+  switch (type) {
+    case "ADD_FAV_BLOG":
+      if (!favBlogs.find((blog) => blog.id === payload.id)) {
+        return { favBlogs: [...favBlogs, payload] };
+      }
+      break;
+    case "REMOVE_FAV_BLOG": {
+      const newFavBlogArr = favBlogs.filter((blog) => blog.id !== payload.id);
+      return { favBlogs: newFavBlogArr };
+    }
+    default: {
+      throw new Error(`Unhandled action type: ${type}`);
+    }
+  }
+}
```

This is beautiful, but isn't this basically Redux...? :) 

If you don't know Redux, this is exactly how it looks. 

Not only that, the `component -> dispatch action -> change reducer -> update component` architecture is exactly what Redux does...

And in my opinion, if we end up using Context like Redux, then we should use Redux, which is well tested and widely used. 

What do you think?

Plus, I don't think that `default` should throw an error.
It supposed just to send back the state (so not to change anything)

---

Really good points. I agree re Redux (we already had this convo in person). 

Updated default to return state. Saw it implemented w/ an error. But what you said makes much more sense.